### PR TITLE
Add ability to get resume data synchronously

### PR DESCRIPTION
**Pull request by glassez**

### CODE REVIEW

**Comment by glassez** [comment]
@arvidn
I don't know if you're notified of new Issues/PRs immediately, or if you review them from time to time, so I preferred to call you explicitly.
I would like to draw your attention to #7855 and get at least a general feedback on it as early as possible. I don't want to be a "train lagger" if it could have been approved, but it was noticed too late. I could contribute to its implementation myself, but I don't want to start work without knowing if it has a chance to be merged.

**Comment by arvidn** [comment]
I think this PR looks good. It would be nice to expose it to python, add a test to exercise it and I don't think you need the `static` empty `add_torrent_params` object, since you're returning it by value. (had the return value been by reference on the other hand, you would have needed it).

**Comment by glassez** [comment]
> Probing the ground in the direction of solving #7855...

Unfortunately, I got my cards mixed up in a hurry. These changes are not really an attempt to solve #7855, but just a quick workaround for some problems related to generating torrent file as mentioned in #7854.

File: include/libtorrent/torrent_handle.hpp [hunk 1]
```diff
@@ -844,6 +845,8 @@ namespace aux {
 		bool need_save_resume_data() const;
 		bool need_save_resume_data(resume_data_flags_t flags) const;
 
+		add_torrent_params get_resume_data(resume_data_flags_t flags = {}) const;
```
Comment (glassez) [inline include/libtorrent/torrent_handle.hpp:848]: @arvidn 
Does it make sense to support flags in this function? The only flag that makes sense when applied to it is `save_info_dict`. But the point is that for some time now it only assign a pointer to `add_torrent_params`. What could be the problem of doing this unconditionally? The caller could just clear it if it is not needed. (IMO, `save_info_dict` could be even deprecated at all.)

**Comment by glassez** [comment]
> I don't think you need the `static` empty `add_torrent_params` object, since you're returning it by value. (had the return value been by reference on the other hand, you would have needed it).

Ok, changed.


> It would be nice to expose it to python

I don't know anything about it at all. At least where is it?

>add a test to exercise it

I could add something similar to `save_resume_data()`, but I found a lot of occurrences of it in the test sources. I doubt what exactly I should do about it.

**Review by arvidn — APPROVED** [review]

**Comment by arvidn** [comment]
I can add documentation and a Changelog entry

**Comment by arvidn** [comment]
https://github.com/arvidn/libtorrent/pull/7972
